### PR TITLE
[MI-2057] Integrate subscription list page

### PR DESCRIPTION
**Pull request by ayusht2810**
image
image
image

### CODE REVIEW

File: webapp/src/containers/Rhs/index.tsx [hunk 1]
```diff
@@ -21,8 +21,8 @@ const Rhs = (): JSX.Element => {
             }
             {
                 usePlugin.isUserAccountConnected() && (
-                    getprojectDetailsState(usePlugin.state).projectID ?
-                        <ProjectDetails title={getprojectDetailsState(usePlugin.state).projectName}/> :
+                    getProjectDetailsState(usePlugin.state).projectID ?
+                        <ProjectDetails {...getProjectDetailsState(usePlugin.state)}/> :
```
Comment (raghavaggarwal2308) [inline webapp/src/containers/Rhs/index.tsx:25]: Create a variable for `getProjectDetailsState(usePlugin.state)`

**Review by raghavaggarwal2308 — CHANGES_REQUESTED** [review]

**Review by raghavaggarwal2308 — APPROVED** [review]

File: webapp/src/containers/Rhs/projectDetails/index.tsx [hunk 2]
```diff
@@ -1,68 +1,163 @@
...
+    // Fetch subscription list
+    const fetchSubscriptionList = () => usePlugin.makeApiRequest(plugin_constants.pluginApiServiceConfigs.getSubscriptionList.apiServiceName, project);
+
     // Reset the state when the component is unmounted
     useEffect(() => {
-        return handleResetProjectDetails();
+        fetchSubscriptionList();
+        return () => {
+            handleResetProjectDetails();
+        };
```
Comment (muazzamqazi) [inline webapp/src/containers/Rhs/projectDetails/index.tsx:83]:
```suggestion
        return handleResetProjectDetails;
```

File: webapp/src/containers/Rhs/projectDetails/index.tsx [hunk 3]
```diff
@@ -1,68 +1,163 @@
...
+                )
+            }
             <div className='d-flex'>
-                <p className='rhs-title'>{title}</p>
+                <p className='rhs-title'>{projectDetails.projectName}</p>
                 <IconButton
                     tooltipText='Unlink project'
                     iconClassName='fa fa-chain-broken'
                     extraClass='project-details-unlink-button unlink-button'
+                    onClick={() => handleUnlinkProject()}
```
Comment (muazzamqazi) [inline webapp/src/containers/Rhs/projectDetails/index.tsx:126]:
```suggestion
                    onClick={handleUnlinkProject}
```

**Review by muazzamqazi — APPROVED** [review]

File: server/plugin/api.go [hunk 4]
```diff
@@ -291,6 +296,13 @@ func (p *Plugin) handleCreateSubscriptions(w http.ResponseWriter, r *http.Reques
 		return
 	}
 
+	channel, channelErr := p.API.GetChannel(body.ChannelID)
+	if channelErr != nil {
+		p.API.LogError("Error in getting channels for team and user", "Error", channelErr.Error())
```
Comment (avas27JTG) [inline server/plugin/api.go:301]: create constant for msgs

File: webapp/src/containers/Rhs/projectDetails/index.tsx [hunk 5]
```diff
@@ -1,68 +1,161 @@
...
+            eventType: subscriptionDetails.eventType,
+            channelID: subscriptionDetails.channelID,
+        });
+        setShowSubscriptionConfirmationModal(true);
+    };
+
+    // Handles unlinking a project and fetching the modified project list
+    const handleConfirmUnlinkProject = async () => {
+        const unlinkProjectStatus = await usePlugin.makeApiRequest(plugin_constants.pluginApiServiceConfigs.unlinkProject.apiServiceName, projectDetails);
+
```
Comment (avas27JTG) [inline webapp/src/containers/Rhs/projectDetails/index.tsx:55]: most of the content of this has been refactored so, leaving those

**Review by avas27JTG — APPROVED** [review]

**Comment by avas27JTG** [comment]
Changes picked here https://github.com/Brightscout/mattermost-plugin-azure-devops/pull/69

**Pull request closed without merging**